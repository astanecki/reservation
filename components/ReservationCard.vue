<template>
  <div class="reservation-card">
    <label class="reservation-card__price">{{ price }} {{ currency }}</label>

    <div class="reservation-card__rating">
      <div class="reservation-card__rating-stars">
        <Stars :number="rating"/>
      </div>
      <div class="reservation-card__rating-quantity">{{ ratingQuantity }}</div>
    </div>

    <div class="reservation-card__divider"></div>

    <slot></slot>
  </div>
</template>
<script>
import Stars from "~/components/Stars";

export default {
  name: 'ReservationCard',

  components: {
    Stars,
  },

  props: {
    price: {
      type: Number,
      required: true,
    },
    currency: {
      type: String,
      required: true,
    },
    rating: {
      type: Number,
      required: true,
    },
    ratingQuantity: {
      type: Number,
      required: true,
    }
  }
}
</script>
<style lang="scss" scoped>
.reservation-card {
  display: flex;
  flex-direction: column;

  width: 330px;
  height: 575px;
  sborder: 1px solid #c6c6c6;
  border-radius: 5px;
  padding: 25px;
  background-color: #e6e6e6;

  &__price {
    font-size: 25px;
    font-weight: 700;
    display: flex;
    margin-bottom: 5px;
  }

  &__rating {
    display: flex;
    font-size: 13px;
    font-weight: 700;
    margin-bottom: 15px;
  }

  &__rating-stars {
    margin-right: 5px;
  }

  &__divider {
    border: 1px solid #c6c6c6;
    margin-bottom: 15px;
  }

  &__rating-quantity {
    font-size: 12px;
    margin-top: 2px;
  }
}
</style>
