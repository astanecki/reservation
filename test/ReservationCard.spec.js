import { mount } from '@vue/test-utils'
import ReservationCard from '@/components/ReservationCard.vue'

describe('ReservationCard', () => {
  const requiredProps = {
    price: 123,
    currency: 'zÅ‚',
    rating: 4,
    ratingQuantity: 123,
  };

  const createWrapper = (propsData = requiredProps) => {
    return mount(ReservationCard, {
      propsData,
    });
  };

  it('is a Vue instance', () => {
    let wrapper = createWrapper();

    expect(wrapper.vm).toBeTruthy();
  });

  it(`should match initial snapshot`, () => {
    let wrapper = createWrapper();

    expect(wrapper.vm.$el).toMatchSnapshot();
  });
})
