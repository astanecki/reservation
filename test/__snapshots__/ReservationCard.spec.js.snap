// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationCard should match initial snapshot 1`] = `
<div
  class="reservation-card"
>
  <label
    class="reservation-card__price"
  >
    123 zł
  </label>
   
  <div
    class="reservation-card__rating"
  >
    <div
      class="reservation-card__rating-stars"
    >
      <div
        class="stars"
      >
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        <span
          class="stars--filled"
        >
          ★
        </span>
        ☆
      </div>
    </div>
     
    <div
      class="reservation-card__rating-quantity"
    >
      123
    </div>
  </div>
   
  <div
    class="reservation-card__divider"
  />
   
</div>
`;
